test.sh

```diff
@@ -10,21 +10,32 @@
 #   of the furthest right failing command or zero if no command failed (o pipefail)
 set -exuo pipefail
 
-require() { command -v "$1" || { echo "error: $1 command required yet absent" ; exit 1 ; } ; }
+# This function will push an action to a stack of items to be done on script exit
+# in the reverse order that they are passed to this function
+trap ":" EXIT
+add_cleanup() {
+	trap -- "$(
+		printf '%s\n' "$1"
+		# get stack is invoked in eval but shellcheck cannot tell since it is indirect
+		# shellcheck disable=SC2317
+		get_stack() { printf '%s\n' "$3"; }
+		eval "get_stack $(trap -p EXIT)"
+	)" EXIT
+}
+
+require() { command -v "$1" > /dev/null || { echo "error: $1 command required yet absent" ; exit 1 ; } ; }
 require curl
-require flake8
-require chcon
-require podman
-require shellcheck
 
-# Reset the tests and mail directories
-sudo rm -rf test email/logs/* email/mail/*
-mkdir -p test
+# Check for shell script style compliance with shellcheck
+./script-lint.sh
 
-# This is a temporary workaround until we properly implement volumes
-chcon -R -t container_file_t email
+# Check python style compliance with flake8
+pushd orbit
```

Comment:
add_cleanup popd

Reply:
no this is incorrect. We need to popd before we can run the other tests and if the check fails we exit because of set -e. If we had a way to do like a new scope that would cleanup immediately ala raii in C++ maybe, but what you are suggesting does not work (it only happened to work in mega_cleanup because I hadn't moved the linting up to before the functional tests)